src/contracts/core/RewardsCoordinator.sol

```diff
@@ -271,6 +293,76 @@ contract RewardsCoordinator is
         _setGlobalOperatorCommission(_globalCommissionBips);
     }
 
+    /// @inheritdoc IRewardsCoordinator
+    function setOperatorAVSCommission(
+        address operator,
+        address avs,
+        uint16 commission
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_AVS_COMMISSION) {
+        require(msg.sender == operator, "RewardsCoordinator.setOperatorAVSCommission: caller is not the operator");
+        require(
+            commission <= ONE_HUNDRED_IN_BIPS,
+            "RewardsCoordinator.setOperatorAVSCommission: commission must be <= 10000 bips"
+        );
+        uint32 activatedAt = SafeCast.toUint32(block.timestamp) + activationDelay;
+
+        OperatorCommission storage operatorCommission = operatorAVSCommissionBips[operator][avs];
+
+        // If, the earlier 'new' commission is activated, we update the 'old' commission with the earlier 'new' commission.
+        // Else, the earlier 'old' commission remains the same. This is essentially resetting the activation delay window
+        // since the earlier commission setting didn't complete.
+        if (block.timestamp >= operatorCommission.activatedAt) {
+            operatorCommission.oldCommissionBips = operatorCommission.newCommissionBips;
+        }
+        operatorCommission.newCommissionBips = commission;
+        operatorCommission.activatedAt = activatedAt;
+
+        emit OperatorAVSCommissionBipsSet(
+            msg.sender,
+            operator,
+            avs,
+            activatedAt,
+            operatorCommission.oldCommissionBips,
+            commission
+        );
+    }
+
+    /// @inheritdoc IRewardsCoordinator
+    function setOperatorPICommission(
```

Comment:
We can dedupe the setting & checking logic with a helper function `_setOperatorCommission(OperatorCommission storage operatorCommission, uint16 newCommission)`

Reply:
Do we foresee a third occurrence of this? If not refactoring might be overkill for the small benefit.

Reply:
Thought about it a bit more and decided to refactor and dedupe it: [9f18c44](https://github.com/Layr-Labs/eigenlayer-contracts/pull/837/commits/9f18c445e548736086ffccd8df7ac57e25433ce5)